components/org.wso2.carbon.identity.oauth.ciba/src/main/java/org/wso2/carbon/identity/oauth/ciba/dao/CibaMgtDAOImpl.java

```diff
@@ -0,0 +1,365 @@
+                    prepStmtForScope.setString(2, singleScopeValue);
+                    prepStmtForScope.addBatch();
+                }
+                prepStmtForScope.executeBatch();
+
+                if (log.isDebugEnabled()) {
+                    log.debug("Successfully persisted scopes for unique authCodeKey : " +
+                            cibaAuthCodeDO.getCibaAuthCodeKey());
+                }
+            } catch (SQLException e) {
+                IdentityDatabaseUtil.rollbackTransaction(connection);
+                throw new CibaCoreException("Error occurred while persisting scopes for the application with " +
+                        "consumer key: " + cibaAuthCodeDO.getConsumerKey(), e);
+            }
+
+            IdentityDatabaseUtil.commitTransaction(connection);
+        } catch (SQLException e) {
+            throw new CibaCoreException("Error occurred while persisting cibaAuthCode for the application with " +
+                    "consumer key: " + cibaAuthCodeDO.getConsumerKey(), e);
+        }
+    }
+
+    @Override
+    public CibaAuthCodeDO getCibaAuthCode(String authCodeKey) throws CibaCoreException {
+
+        CibaAuthCodeDO cibaAuthCodeDO = new CibaAuthCodeDO();
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(false)) {
+            try (PreparedStatement prepStmt = connection.prepareStatement(SQLQueries.
+                    CibaSQLQueries.RETRIEVE_AUTH_CODE)) {
+
+                prepStmt.setString(1, authCodeKey);
+                ResultSet resultSet = prepStmt.executeQuery();
+                if (resultSet.next()) {
+                    cibaAuthCodeDO.setCibaAuthCodeKey(resultSet.getString(1));
+                    cibaAuthCodeDO.setAuthReqID(resultSet.getString(2));
+                    cibaAuthCodeDO.setConsumerKey(resultSet.getString(3));
+                    cibaAuthCodeDO.setLastPolledTime(
+                            resultSet.getTimestamp(4, Calendar.getInstance(TimeZone.getTimeZone(CibaConstants.UTC))));
+                    cibaAuthCodeDO.setInterval(resultSet.getLong(5));
+                    cibaAuthCodeDO.setExpiresIn(resultSet.getLong(6));
+                    cibaAuthCodeDO.setAuthReqStatus(AuthReqStatus.valueOf(resultSet.getString(7)));
+                    cibaAuthCodeDO.setIssuedTime(
+                            resultSet.getTimestamp(8, Calendar.getInstance(TimeZone.getTimeZone(CibaConstants.UTC))));
+                } else {
+                    return null;
+                }
+
+                if (log.isDebugEnabled()) {
+                    log.debug("Successfully obtained cibaAuthCode for unique authCodeKey : " + authCodeKey);
+                }
+            }
+        } catch (SQLException e) {
+            throw new CibaCoreException("Error in obtaining cibaAuthCode for unique authCodeKey : " + authCodeKey, e);
+        }
+        return cibaAuthCodeDO;
+    }
+
+    @Override
+    public String[] getScope(String authCodeKey) throws CibaCoreException {
```

Comment:
Shall we return a list from this method

Reply:
Is that you mean scopes as a list from this method only or all the places where scope [ acr,audience too] defined as String array.